crates/runtime/src/model/chat.rs

```diff
@@ -69,18 +69,28 @@ pub async fn try_to_chat_model<S: ::std::hash::BuildHasher>(
     let model = construct_model(&prefix, model_id, component, params)?;
 
     // Handle tool usage
-    let spice_tool_opt: Option<SpiceToolsOptions> = params
-        .get("spice_tools")
-        .map(Secret::expose_secret)
+    let spice_tool_opt: Option<SpiceToolsOptions> = extract_secret!(params, "spice_tools")
```

Comment:
There should be an intelligent default set somewhere.

Reply:
Currently the `"intelligent"` default is the `auto` keyword.
```yaml
params:
  spice_tools: auto
```